gossipd/broadcast.h

```diff
@@ -37,9 +37,13 @@ struct broadcast_state *new_broadcast_state(struct routing_state *rstate,
 					    struct list_head *peers);
 
 /* Append a queued message for broadcast.  Must be explicitly deleted.
- * Also adds it to the gossip store. */
+ * Also adds it to the gossip store.
+ *
+ * If it's a channel_announcement, channel_announce_sat must be set.
+ */
 void insert_broadcast(struct broadcast_state **bstate,
 		      const u8 *msg,
+		      const struct amount_sat *channel_announce_sat,
```

Comment:
I have a bit of an aversion against this kind of argument whose existence/value depend on other arguments.

Reply:
I do agree that freeing `gossip_store` from having to look up external information is good here, but since we're driving this from outside anyway, why not just have the caller build the channel amount msg and store that?

Reply:
I have done this, but it's not as trivially neat as I'd like.  Now routing.c creates a gossip_store msg, so I don't think we won.

The alternative was to append the channel size to the channel_announcement itself, and have everyone "know" to remove that.  Neither is perfect.